### PR TITLE
Use the EIA API to update all lands/waters production data

### LINKED ISSUE #1766: Update all-lands production data (EIA)
We have 2013 info from EIA. We should update it to the latest available.

### PR DESCRIPTION
**Pull request by shawnbot**
Fixes #1766.

[:sunglasses: Preview: National](https://federalist.18f.gov/preview/18F/doi-extractives-data/all-prod-eia-api/states/#all-production)
[:sunglasses: Preview: Wyoming](https://federalist.18f.gov/preview/18F/doi-extractives-data/all-prod-eia-api/states/WY/#all-production)

Changes proposed in this pull request:
- [x] replaces the old EIA data downloads with better, more up-to-date data
- [x] introduces an update script to fetch new data from the EIA API
- [x] modifies some of the renewables commodities:
  * `Biomass (total)` is gone; the EIA no longer lists this category of renewables production
  * `Oil` is now, more specifically, `Crude Oil`.
  * Likewise, `Gas` is now specifically `Natural Gas`.
  * ~~Adds a [renewable energy total](http://www.eia.gov/opendata/qb.cfm?sdid=SEDS.REPRB.US.A), measured by EIA in billions of btu~~
- [x] Updates the SQL data import scripts to use the new data
- [x] Updates the SQL rollup script accordingly
- [x] Updates the `all_state_production` and `all_national_production` YAML data
- [x] Add more helpful error messages to the update script when environment vars aren't set
- [x] Update the data tests to ensure integrity of the new data
- [x] Get a review from @gemfarmer

I'm going to save the remaining work on offshore production data (#1771) for a future PR.

/cc @gemfarmer @coreycaitlin

### CODE REVIEW

**Comment by coreycaitlin** [comment]
Found it! Here's where we figured out what to show and what not to show: #1574

**Comment by coreycaitlin** [comment]
@shawnbot it looks like we've lost some of our units!

screen shot 2016-08-25 at 9 51 48 am

**Comment by coreycaitlin** [comment]
@shawnbot aha, maybe this is only showing up as a problem on the national page?

File: Makefile [hunk 1]
```diff
@@ -557,43 +562,15 @@ tables/federal_offshore_production: data/federal-production/offshore.tsv
...
+	@$(call drop-table,all_production)
+	tmp=$^/all.ndjson; \
+	for tsv in $^/*.tsv; do \
+		$(tito) -r tsv $$tsv >> $$tmp; \
 	done; \
-	$(tables) -t ndjson -n all_production_coal -i $$tmp; \
+	$(tables) -t ndjson -n all_production -i $$tmp && \
 	rm $$tmp
-
-tables/all_production_oil: data/all-production/input/oil.tsv
```
Comment (gemfarmer) [inline Makefile]: This is soooo much better!

**Comment by gemfarmer** [comment]
This looks really great, @shawnbot! And it comes with data tests. Merge at will :)

**Pull request merged**